tests/components/guardian/test_diagnostics.py

```diff
@@ -4,15 +4,16 @@
 from homeassistant.components.guardian import DOMAIN, GuardianData
 from homeassistant.core import HomeAssistant
 
+from tests.common import MockConfigEntry
 from tests.components.diagnostics import get_diagnostics_for_config_entry
 from tests.typing import ClientSessionGenerator
 
 
 async def test_entry_diagnostics(
     hass: HomeAssistant,
-    config_entry,
+    config_entry: MockConfigEntry,
     hass_client: ClientSessionGenerator,
-    setup_guardian,
+    setup_guardian: None,
 ) -> None:
     """Test config entry diagnostics."""
     data: GuardianData = hass.data[DOMAIN][config_entry.entry_id]
```

Comment:
Side note for the code owner => this could be migrated to use `config_entry.runtime_data` :)